bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/PatientDaoImpl.java

```diff
@@ -100,6 +105,66 @@ public PatientDaoImpl(SessionFactory sessionFactory) {
         return patientResponses;
     }
 
+    @Override
+    public List<PatientResponse> getSimilarPatientsUsingLuceneSearch(String identifier, String name, String gender, String customAttribute,
+                                                              String addressFieldName, String addressFieldValue, Integer length,
+                                                              Integer offset, String[] customAttributeFields, String programAttributeFieldValue,
```

Comment:
I didn't look closely but I think you're ignoring the length and offset parameters.